graphene_django/fields.py

```diff
@@ -135,22 +135,23 @@ def resolve_connection(cls, connection, args, iterable, max_limit=None):
 
         if isinstance(iterable, QuerySet):
             list_length = iterable.count()
-            list_slice_length = (
-                min(max_limit, list_length) if max_limit is not None else list_length
-            )
         else:
             list_length = len(iterable)
-            list_slice_length = (
-                min(max_limit, list_length) if max_limit is not None else list_length
-            )
+        list_slice_length = (
+            min(max_limit, list_length) if max_limit is not None else list_length
+        )
 
         # If after is higher than list_length, connection_from_list_slice
         # would try to do a negative slicing which makes django throw an
         # AssertionError
         after = min(get_offset_with_default(args.get("after"), -1) + 1, list_length)
 
         if max_limit is not None and "first" not in args:
-            args["first"] = max_limit
+            if "last" in args:
+                args["first"] = list_length
```

Comment:
Wondering why is this needed? In #1074 I didn't have to include it, seems like removing it wouldn't change anything.

Reply:
I would have to check if it is actually used by `connection_from_list_slice` but I set it at least for consistency with the other case where it is set to `max_limit`